core/trino-parser/src/main/java/io/trino/sql/tree/JsonExists.java

```diff
@@ -0,0 +1,181 @@
+        this.inputFormat = inputFormat;
+        this.jsonPath = jsonPath;
+        this.pathParameters = ImmutableList.copyOf(pathParameters);
+        this.errorBehavior = errorBehavior;
+    }
+
+    public enum ErrorBehavior
+    {
+        FALSE, // default
+        TRUE,
+        UNKNOWN,
+        ERROR
+    }
+
+    public Expression getInputExpression()
+    {
+        return inputExpression;
+    }
+
+    public JsonFormat getInputFormat()
+    {
+        return inputFormat;
+    }
+
+    public StringLiteral getJsonPath()
+    {
+        return jsonPath;
+    }
+
+    public List<JsonPathParameter> getPathParameters()
+    {
+        return pathParameters;
+    }
+
+    public ErrorBehavior getErrorBehavior()
+    {
+        return errorBehavior;
+    }
+
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context)
+    {
+        return visitor.visitJsonExists(this, context);
+    }
+
+    @Override
+    public List<? extends Node> getChildren()
+    {
+        ImmutableList.Builder<Node> children = ImmutableList.builder();
+        children.add(inputExpression);
+        pathParameters.stream()
+                .map(JsonPathParameter::getParameter)
+                .forEach(children::add);
+        return children.build();
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        JsonExists that = (JsonExists) o;
+        return Objects.equals(inputExpression, that.inputExpression) &&
+                inputFormat == that.inputFormat &&
+                Objects.equals(jsonPath, that.jsonPath) &&
+                Objects.equals(pathParameters, that.pathParameters) &&
+                errorBehavior == that.errorBehavior;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(inputExpression, inputFormat, jsonPath, pathParameters, errorBehavior);
+    }
+
+    @Override
+    public boolean shallowEquals(Node other)
+    {
+        if (!sameClass(this, other)) {
+            return false;
+        }
+
+        JsonExists otherJsonExists = (JsonExists) other;
+        boolean parameterNamesEqual = Objects.equals(
+                pathParameters.stream()
+                        .map(JsonPathParameter::getName)
+                        .collect(toImmutableList()),
+                otherJsonExists.pathParameters.stream()
+                        .map(JsonPathParameter::getName)
+                        .collect(toImmutableList()));
+        boolean parameterFormatsEqual = Objects.equals(
+                pathParameters.stream()
+                        .map(JsonPathParameter::getFormat)
+                        .collect(toImmutableList()),
+                otherJsonExists.pathParameters.stream()
+                        .map(JsonPathParameter::getFormat)
+                        .collect(toImmutableList()));
```

Comment:
This should not be part of shallowEquals as they are nested tree nodes. `getChildren` should just return the list of JsonPathParameter entries, as that represents the syntax tree hierarchy.